### PR TITLE
Updates for new type system and tuple-printing

### PR DESCRIPTION
**Pull request by timholy**
This fixes several breakages on julia 0.6. ~~Needs https://github.com/JuliaLang/julia/pull/20568 and~~ affected by https://github.com/JuliaLang/julia/issues/20569. But at least this way we're better prepared.

### CODE REVIEW

File: src/ImageCore.jl [hunk 1]
```diff
@@ -4,6 +4,7 @@ module ImageCore
 
 using Colors, FixedPointNumbers, MappedArrays, Graphics, ShowItLikeYouBuildIt
 using OffsetArrays # for show.jl
+using ColorTypes: colorant_string
```
Comment (tkelman) [inline src/ImageCore.jl:7]: should be listed in REQUIRE if used directly

Comment (timholy) [reply src/ImageCore.jl:7]: Indeed it should. Damn, you're impressive.

Comment (tkelman) [reply src/ImageCore.jl:7]: observant, picky, OCD perhaps

this direct dependency stuff will hopefully be baked into the way Pkg3 environments work so it'll get auto-detected

**Comment by timholy** [comment]
Switched to a workaround for https://github.com/JuliaLang/julia/pull/20568.

**Comment by timholy** [comment]
This requires a new tag in OffsetArrays to pass tests, and at least some versions of 0.6 have an inference problem with the StackedViews tests (not our CI, apparently). But, this allows ImageCore to at least build on 0.6 and perform most of its duties, which should end the next main reason the whole Images suite is not passing on 0.6.

**Pull request merged**